pisek/task_jobs/generator/pisek_gen_v1.py

```diff
@@ -0,0 +1,144 @@
+        return input_infos
+
+    def _get_input_info_from_line(self, line: str, line_number: int) -> InputInfo:
+        line = line.rstrip("\n")
+        if not line:
+            self._complain_about_line(line_number, line, "line empty")
+
+        args = line.split(" ")
+        input_name = args[0]
+        info_args: dict[str, Any] = {}
+
+        for arg in args[1:]:
+            if "=" not in arg:
+                self._complain_about_line(line_number, line, "missing '='")
+            parts = arg.split("=")
+            if len(parts) != 2:
+                self._complain_about_line(line_number, line, "too many '='")
+            arg_name, arg_value = parts
+            if arg_name == "repeat":
+                if not arg_value.isnumeric():
+                    self._complain_about_line(
+                        line_number, line, "repeat should be a natural number"
+                    )
+                info_args[arg_name] = int(arg_value)
+            elif arg_name == "seeded":
+                if arg_value not in ("true", "false"):
+                    self._complain_about_line(
+                        line_number, line, "seeded should be 'true' or 'false'"
+                    )
+                info_args[arg_name] = arg_value == "true"
+            else:
+                self._complain_about_line(
+                    line_number, line, f"unknown argument: {arg_name}"
+                )
+
+        return InputInfo.generated(input_name, **info_args)
+
+    def _complain_about_line(self, line_index: int, contents: str, reason: str) -> None:
+        message = (
+            f"{self.generator}: invalid line {line_index} of listed inputs:\n"
+            f"{contents}\n"
+            f"(reason: {reason})"
+        )
+        raise self._create_program_failure(message, self._run_result)
+
+    def _get_input_lines(self) -> list[str]:
+        inputs_list_path = TaskPath.data_path(self._env, "inputs_list")
+
+        self._run_result = self._run_program(
+            ProgramType.in_gen,
+            self.generator,
+            stdout=inputs_list_path,
+            stderr=TaskPath.log_file(self._env, "inputs_list", self.generator.name),
+        )
+
+        if self._run_result.kind != RunResultKind.OK:
+            raise self._create_program_failure(
+                f"{self.generator} failed to list inputs",
+                self._run_result,
+            )
+
+        with self._open_file(inputs_list_path) as f:
+            return f.readlines()
+
+
+class PisekGenV1GeneratorJob(ProgramsJob):
+    """Abstract class for jobs with OnlineGenerator."""
+
+    generator: TaskPath
+    seed: int
+    input_info: InputInfo
+    input_path: TaskPath
+
+    def __init__(self, env: Env, *, name: str = "", **kwargs) -> None:
+        super().__init__(env=env, name=name, **kwargs)
+
+    def _gen(self) -> None:
+        if self.seed < 0:
+            raise ValueError(f"Seed {self.seed} is negative.")
```

Comment:
Error messages usually don't start with a capital letter and don't end with a period.

Reply:
I have also fixed this in the opendata_v1 generator.